### PR TITLE
For 4.0: Require all stable feature flags added up to 3.13.0

### PR DESCRIPTION
**Pull request by ansd**
Since feature flag `message_containers` introduced in 3.13.0 is required in 4.0, we can also require all other feature flags introduced in or before 3.13.0 and remove their compatibility code for 4.0:

* `restart_streams`
* `stream_sac_coordinator_unblock_group`
* `stream_filtering`
* `stream_update_config_command`

Related to https://github.com/rabbitmq/rabbitmq-server/pull/11642

### CODE REVIEW

File: deps/rabbit/src/rabbit_stream_queue.erl [hunk 1]
```diff
@@ -96,8 +96,7 @@
                         soft_limit :: non_neg_integer(),
                         slow = false :: boolean(),
                         readers = #{} :: #{rabbit_types:ctag() => #stream{}},
-                        writer_id :: binary(),
-                        filtering_supported :: boolean()
```
Comment (dumbbell) [inline deps/rabbit/src/rabbit_stream_queue.erl:100]: Are we 100% sure that this record is never stored in Mnesia/Khepri and never sent to another node?

Comment (ansd) [reply deps/rabbit/src/rabbit_stream_queue.erl:100]: yes

File: deps/rabbit/test/rabbit_stream_queue_SUITE.erl [hunk 2]
```diff
@@ -1256,7 +1239,7 @@ recover_after_leader_and_coordinator_kill(Config) ->
 
 
     ct:pal("sys state ~p", [CState]),
-
+    rabbit_ct_broker_helpers:rpc(Config, 0, ?MODULE, delete_testcase_queue, [Q]),
```
Comment (dumbbell) [inline deps/rabbit/test/rabbit_stream_queue_SUITE.erl:1242]: Is this a drive-by change? If this is a fix, should it be backported separately of this whole patch?

File: deps/rabbit/test/rabbit_stream_queue_SUITE.erl [hunk 3]
```diff
@@ -2218,6 +2194,7 @@ leader_locator_balanced_maintenance(Config) ->
       end, 60000),
 
     true = rabbit_ct_broker_helpers:unmark_as_being_drained(Config, Server3),
+    rabbit_ct_broker_helpers:rpc(Config, 0, ?MODULE, delete_queues, [[Q1, Q]]),
```
Comment (dumbbell) [inline deps/rabbit/test/rabbit_stream_queue_SUITE.erl:2197]: Same question, drive-by change?

File: deps/rabbit/src/rabbit_stream_sac_coordinator.erl [hunk 4]
```diff
@@ -629,32 +629,12 @@ handle_consumer_removal(Group0, Consumer, Stream, ConsumerName) ->
...
-        false ->
-            tuple
-    end.
-
 notify_consumer_effect(Pid, SubId, Stream, Name, Active) ->
     notify_consumer_effect(Pid, SubId, Stream, Name, Active, false).
 
 notify_consumer_effect(Pid, SubId, Stream, Name, Active, SteppingDown) ->
-    notify_consumer_effect(Pid, SubId, Stream, Name, Active, SteppingDown, message_type()).
+    notify_consumer_effect(Pid, SubId, Stream, Name, Active, SteppingDown, map).
```
[the remark this answers is not in the document]
Comment (ansd) [reply deps/rabbit/src/rabbit_stream_sac_coordinator.erl]: Good spot, thanks!

File: deps/rabbit/test/rabbit_stream_queue_SUITE.erl (hunk 2 above)
Comment (ansd) [reply deps/rabbit/test/rabbit_stream_queue_SUITE.erl:1242]: As I already wrote in Slack:
> Test cases rebalance and restart_stream were always skipped prior to this PR because the feature flag restart_streams in these test cases is misspelled. So, that part should be backported to v3.13.x

This is not a "drive-by change" because this PR removes the code to enable the feature flag in https://github.com/rabbitmq/rabbitmq-server/pull/11659/files#diff-bd1bb71aee8de223a400a433496cefbd338052929a9ed71906665ca953b72746L2820-L2827 which causes test case `rebalance` to run for the first time. The test case was buggy. One bug is the line you are commenting: Test cases didn't clean up properly after each other which caused assertions in test case `rebalance` to fail. (In general, I prefer to randomise test case executions within a group such that test interference can be easily spotted.) So, these test case fixes are already in a separate commit in this PR and can be easily backported to `v3.13.x`.

File: deps/rabbit/test/rabbit_stream_queue_SUITE.erl (hunk 3 above)
Comment (ansd) [reply deps/rabbit/test/rabbit_stream_queue_SUITE.erl:2197]: same answer as above

File: deps/rabbit/test/rabbit_stream_queue_SUITE.erl (hunk 2 above)
Comment (ansd) [reply deps/rabbit/test/rabbit_stream_queue_SUITE.erl:1242]: Here is the commit I'm referring to: https://github.com/rabbitmq/rabbitmq-server/pull/11659/commits/6cdfb62a81c287f017a0b8907ee98ab759430583

**Review by dumbbell — APPROVED** [review]

File: deps/rabbit/test/rabbit_stream_queue_SUITE.erl (hunk 2 above)
Comment (dumbbell) [reply deps/rabbit/test/rabbit_stream_queue_SUITE.erl:1242]: Thank you, I understand now.

**Review by acogoluegnes — APPROVED** [review]
